### PR TITLE
Add incorporation modal with slides

### LINKED ISSUE #4154: Incorporation: As a user I can see Modal with Slides, to learn about DAO incorporation
[Provided by Adrian via Contractors project management platform]

## Description
To help users understand they purpose of incorporation, why it is important, the process, and the cost; we would like to have a stepped process prior to starting the incorporation process that includes necessary details. At the end of the process they will be able to continue onto start the actual incorporation.

## Acceptance Criteria

- an option to see modal with 3 slides
- an option to close modal with (x) icon
- an option to see heading "About incorporation"
- an option to see the titles of slides and vertical bar (colour of the bar should change on active state)
- an option to navigate slides using "Next"/"Previous" button
  - "Previous" button should not be visible on the first slide
  - "Incorporate" button should be visible on the last slide, instead of "Next" button
- an option to see content of the slides:
  - an option to see heading
  - an option to see text area
  - an option to see labels/steps
  - an option to see costs

## Design

- [Figma Link - Slide 1](https://www.figma.com/file/fmnKIUY0LGfEnsOOQloVVX/Korporatio?node-id=976%3A109822&t=2F2oGhbRnHI53L7V-0)

image

- [Figma Link - Slide 1](https://www.figma.com/file/fmnKIUY0LGfEnsOOQloVVX/Korporatio?node-id=976%3A131168&t=2F2oGhbRnHI53L7V-0)

image

- [Figma Link - Slide 1](https://www.figma.com/file/fmnKIUY0LGfEnsOOQloVVX/Korporatio?node-id=976%3A131409&t=2F2oGhbRnHI53L7V-0)

image

### PR DESCRIPTION
**Pull request by agnieszkajarosikloj**
## Description

This PR adds Modal with Slides, to learn about DAO incorporation. Modal consists of 3 tabs and slides. User can switch between slides by clicking on the tab or on the button. 

To test click on the New Action button and choose DAO Incorporation option.

**New stuff** ✨

* Modal with slides about DAO incorporation.

**Changes** 🏗

* Added **widthAuto** prop to the Dialog core component. With this flag set to true, the dialog can be wider than 500 pixels.
Screenshot 2023-01-26 at 23 04 28

[Figma Link - Slide 1](https://www.figma.com/file/fmnKIUY0LGfEnsOOQloVVX/Korporatio?node-id=976%3A109822&t=2F2oGhbRnHI53L7V-0)
[Figma Link - Slide 2](https://www.figma.com/file/fmnKIUY0LGfEnsOOQloVVX/Korporatio?node-id=976%3A131168&t=2F2oGhbRnHI53L7V-0)
[Figma Link - Slide 3](https://www.figma.com/file/fmnKIUY0LGfEnsOOQloVVX/Korporatio?node-id=976%3A131409&t=2F2oGhbRnHI53L7V-0)

Resolves  #4154

### CODE REVIEW

File: src/modules/dashboard/components/Dialogs/DAOIncorporationDialog/TabPanels/HowPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    id: 'dashboard.DAOIncorporationDialog.TabPanels.HowPanel.step3',
+    defaultMessage: `With the application open, collective discussion can happen to agree to all the details, nominated Protectors, and the cost to incorporate. Protectors will also need to submit verification details in the step.<div> Step time: Dependent on parties involved </div>`,
+  },
+  step4: {
+    id: 'dashboard.DAOIncorporationDialog.TabPanels.HowPanel.step4',
+    defaultMessage: `Create a Motion to pay for the incorporation. This is the final step to get collective approval of the decision.<div> Step time: Dependent on Motion duration </div>`,
+  },
+  step5: {
+    id: 'dashboard.DAOIncorporationDialog.TabPanels.HowPanel.step5',
+    defaultMessage: `Korporatio will process the application and finalise the incorporate. The DAO will then be able to start benefitting for a legal DAO wrapper.<div> Step time: 15 - 20 business days </div>`,
```
Comment (arrenv) [inline src/modules/dashboard/components/Dialogs/DAOIncorporationDialog/TabPanels/HowPanel.tsx:35]: There is a type here, please change to: "Korporatio will process the application and finalize the incorporation. The DAO and contributors will then start benefiting from the legal DAO wrapper.<div> Step time: 15 - 20 business days"

File: src/modules/dashboard/components/Dialogs/DAOIncorporationDialog/TabPanels/WhyPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import styles from './TabPanels.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.DAOIncorporationDialog.TabPanels.WhyPanel.title',
+    defaultMessage: 'Why incorporate your DAO',
+  },
+  description: {
+    id: 'dashboard.DAOIncorporationDialog.TabPanels.WhyPanel.description',
+    defaultMessage: `Setup a legal wrapper for you DAO to extend it’s functionality and help protect contributors. Incorporation is done in Panama through a partner of Colony, Korporatio. Who is specifically focused on helping to support smart companies built and run on-chain.`,
```
Comment (arrenv) [inline src/modules/dashboard/components/Dialogs/DAOIncorporationDialog/TabPanels/WhyPanel.tsx:16]: Please change to: "Setup a legal wrapper for you DAO to extend it’s functionality and help protect contributors. Incorporation is done in Panama through a service provided by Korporatio. Who is specifically focused on helping to support smart companies built and run on-chain."

**Review by arrenv — CHANGES_REQUESTED** [review]
Thanks @agnieszkajarosikloj, this has the same issue as https://github.com/JoinColony/colonyDapp/pull/4141 regarding Incorporating being in the wrong location of the modal.

**Comment by agnieszkajarosikloj** [comment]
@arrenv  I added changes. Can you have a look?

File: src/modules/dashboard/components/Dialogs/DAOIncorporationDialog/TabPanels/WhyPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import styles from './TabPanels.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.DAOIncorporationDialog.TabPanels.WhyPanel.title',
+    defaultMessage: 'Why incorporate your DAO',
+  },
+  description: {
+    id: 'dashboard.DAOIncorporationDialog.TabPanels.WhyPanel.description',
+    defaultMessage: `Setup a legal wrapper for you DAO to extend it's functionality and help protect contributors. Incorporation is done in Panama through a service provided by Korporatio. Who is specifically focused on helping to support smart companies built and run on-chain.`,
```
Comment (arrenv) [inline src/modules/dashboard/components/Dialogs/DAOIncorporationDialog/TabPanels/WhyPanel.tsx:16]: My apologies, I have a typo in my replacement, it should be: "Setup a legal wrapper for your DAO to extend it's functionality and help protect contributors. Incorporation is done in Panama through a service provided by Korporatio. Who is specifically focused on helping to support smart companies built and run on-chain."

**Review by arrenv — APPROVED** [review]
Just one typo on my end, but I have approved.

**Pull request merged**